jwst/ramp_fitting/ramp_fit.py

```diff
@@ -807,172 +1177,150 @@ def gls_ramp_fit(model,
-            s_mask = (gdq_sect[0] == saturated_flag)
-            if s_mask.any():
-                temp_dq[rlo:rhi, :][s_mask] = dqflags.pixel['UNRELIABLE_SLOPE']
-            slope_err_int[num_int, rlo:rhi, :] = np.sqrt(slope_var_sect)
-
-            # We need to take a weighted average if (and only if) n_int > 1.
-            # Accumulate sum of slopes and sum of weights.
-            if n_int > 1:
-                weight = 1. / slope_var_sect
-                slopes[rlo:rhi, :] += (slope_sect * weight)
-                sum_weight[rlo:rhi, :] += weight
-
-            if save_opt:
-                # Save the intercepts and cosmic-ray amplitudes for the
-                # current integration.
-                intercept_int[num_int, rlo:rhi, :] = intercept_sect.copy()
-                intercept_err_int[num_int, rlo:rhi, :] = \
-                        np.sqrt(np.abs(intercept_var_sect))
-                pedestal_int[num_int, rlo:rhi, :] = \
-                        utils.gls_pedestal(first_group[rlo:rhi, :],
-                                           slope_int[num_int, rlo:rhi, :],
-                                           s_mask,
-                                           frame_time, nframes_used)
-                ampl_int[num_int, rlo:rhi, :, :] = cr_sect.copy()
-                ampl_err_int[num_int, rlo:rhi, :, :] = \
-                        np.sqrt(np.abs(cr_var_sect))
-            del s_mask
-
-            # Compress 4D->2D dq arrays for saturated and jump-detected
-            #   pixels
-            pixeldq_sect = pixeldq[rlo:rhi, :].copy()
-            dq_int[num_int, rlo:rhi, :] = \
-                  dq_compress_sect(gdq_sect, pixeldq_sect).copy()
-
-        # temp_dq |= dq_int[num_int, :, :]
-        # dq_int[num_int, :, :] = temp_dq.copy()
-        dq_int[num_int, :, :] |= temp_dq
-        temp_dq[:, :] = 0               # initialize for next integration
-
-    # Average the slope over all integrations.
+            # Create arrays for the fitted values of zero-point intercept and
+            # cosmic-ray amplitudes, and their errors.
+            intercept_int = np.zeros((n_int,) + imshape, dtype=np.float32)
+            intercept_err_int = np.zeros((n_int,) + imshape, dtype=np.float32)
+            # The pedestal is the extrapolation of the first group back to zero
+            # time, for each integration.
+            pedestal_int = np.zeros((n_int,) + imshape, dtype=np.float32)
+            # If there are no cosmic rays, set the last axis length to 1.
+            shape_ampl = (n_int, imshape[0], imshape[1], max(1, max_num_cr))
+            ampl_int = np.zeros(shape_ampl, dtype=np.float32)
+            ampl_err_int = np.zeros(shape_ampl, dtype=np.float32)
+
+##Loop over number of processes
+        for i in range(number_slices - 1):
+            readnoise_slice = readnoise_2d[i * rows_per_slice: (i + 1) * rows_per_slice, :]
+            gain_slice = gain_2d[i * rows_per_slice: (i + 1) * rows_per_slice, :]
```

Comment:
Common indices duplicated: perhaps this can be simplified.

Reply:
I've updated the all of the result indices. If you thing the new way is clearer. I can update the input indices in a similar way